### PR TITLE
use uni v3 for aave lever/delever

### PR DESCRIPTION
**Pull request by jalbrekt85**
handles #658

levering:
```py
SAFE.aave.deposit(WBTC, 5e8)
SAFE.aave.lever_up(WBTC, USDC, .5)
```
sushiswap:
```
              balance_before         balance_after         balance_delta
symbol                                                                  
aWBTC   0.000000000000000000  7.502991230000000000  7.502991230000000000
```
uni v3:
```
              balance_before         balance_after         balance_delta
symbol                                                                  
aWBTC   0.000000000000000000  7.521543920000000000  7.521543920000000000 
```

### CODE REVIEW

File: great_ape_safe/ape_api/aave.py [hunk 1]
```diff
@@ -147,14 +147,13 @@ def lever_up(self, collateral_token, borrow_token, perc):
...
-            we_want_to_borrow,
-            [borrow_token, registry.eth.treasury_tokens.WETH, collateral_token]
-        )[-1]
+        self.safe.init_uni_v3()
+        weth = self.safe.contract(registry.eth.treasury_tokens.WETH)
+        to_reinvest = self.safe.uni_v3.swap(
+            [borrow_token, weth, collateral_token],
+            we_want_to_borrow
+        )
+
```
Comment (gosuto-inzasheru) [inline great_ape_safe/ape_api/aave.py:156]:
```suggestion
        # swap borrowed for more `collateral_token` and deposit back into aave
        self.safe.init_uni_v3()
        to_reinvest = self.safe.uni_v3.swap(
            [borrow_token, collateral_token], we_want_to_borrow
        )
```
no need to go through weth here! we did that on sushi due to it not having a direct wbtc/usdc pool but that is very different on univ3 :)

File: great_ape_safe/ape_api/aave.py [hunk 2]
```diff
@@ -189,13 +188,13 @@ def delever(self, collateral_token, borrow_token):
...
 
-            ## Swap to debt
-            self.safe.init_sushi()
-            to_repay = self.safe.sushi.swap_tokens_for_tokens(
-                collateral_token,
+            # Swap to debt
+            self.safe.init_uni_v3()
+            weth = self.safe.contract(registry.eth.treasury_tokens.WETH)
+            to_repay = self.safe.uni_v3.swap(
+                [collateral_token, weth, borrow_token],
```
Comment (gosuto-inzasheru) [inline great_ape_safe/ape_api/aave.py:195]:
```suggestion
            to_repay = self.safe.uni_v3.swap(
                [collateral_token, borrow_token],
```

File: great_ape_safe/ape_api/aave.py [hunk 3]
```diff
@@ -204,11 +203,11 @@ def delever(self, collateral_token, borrow_token):
         bal_borrow_token_after = borrow_token.balanceOf(self.safe)
 
         ## Swap remaining margin of borrow token back into collateral token
-        self.safe.init_sushi()
-        self.safe.sushi.swap_tokens_for_tokens(
-            borrow_token,
+        self.safe.init_uni_v3()
+        weth = self.safe.contract(registry.eth.treasury_tokens.WETH)
+        self.safe.uni_v3.swap(
+            [borrow_token, weth, collateral_token],
```
Comment (gosuto-inzasheru) [inline great_ape_safe/ape_api/aave.py:209]:
```suggestion
        self.safe.uni_v3.swap(
            [borrow_token, collateral_token],
```

File: great_ape_safe/ape_api/aave.py (hunk 1 above)
Comment (jalbrekt85) [reply great_ape_safe/ape_api/aave.py:156]: good point

**Review by gosuto-inzasheru — APPROVED** [review]

**Review by petrovska-petro — APPROVED** [review]
restructuring to leverage univ3 class looks good ✅

**Pull request merged**